senior_teacher.rb

```diff
@@ -1,39 +1,46 @@
-class SeniorTeacher
+require_relative 'staff'
+
+class SeniorTeacher < Staff
   attr_reader :age, :salary, :phase, :performance_rating, :target_raise
   attr_accessor :name
+  RESPONSE1 = "Listen, class, this is how everything works, fo SHO! "
+  RESPONSE2 = "*drops flat-out insane knowledge bomb* "
+  RESPONSE3 = "... You're welcome. *saunters away*"
+  # TARGET_RAISE = 1000  
 
   def initialize(options={})
-    @phase = 3
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
+    # @phase = 3
+    # @age = options.fetch(:age, 0)
+    # @name = options.fetch(:name, "")
+    super
     @target_raise = 1000
   end
 
-  def offer_high_five
-    "High five!"
-  end
+  # def offer_high_five
+  #   "High five!"
+  # end
 
-  def set_phase(num)
-    @phase = num
-    "Cool, I've always wanted to teach phase #{num}!"
-  end
+  # def set_phase(num)
+  #   @phase = num
+  #   "Cool, I've always wanted to teach phase #{num}!"
+  # end
 
-  def teach_stuff
-    response = ""
-    response += "Listen, class, this is how everything works, fo SHO! "
-    response += "*drops flat-out insane knowledge bomb* "
-    response += "... You're welcome. *saunters away*"
-    response
-  end
+  # def teach_stuff
+  #   response = ""
+  #   response += "Listen, class, this is how everything works, fo SHO! "
+  #   response += "*drops flat-out insane knowledge bomb* "
+  #   response += "... You're welcome. *saunters away*"
+  #   response
+  # end
 
-  def salary=(new_salary)
-    puts "This better be good!"
-    @salary = new_salary
-  end
+  # def salary=(new_salary)
+  #   puts "This better be good!"
+  #   @salary = new_salary
+  # end
 
-  def receive_raise(raise)
-    @salary += raise
-  end
+  # def receive_raise(raise)
+  #   @salary += raise
+  # end
 
   def set_performance_rating(rating)
```

Comment:
This is very similar to the one in ApprenticeTeacher, how could you refactor it?